app/src/test/java/org/fossasia/openevent/app/unit/utils/ErrorUtilsTest.java

```diff
@@ -0,0 +1,139 @@
+    private static final Response<Account> ERROR_RESPONSE_3 = Response.error(422, RESPONSE_BODY_3);
+
+    private static final ResponseBody RESPONSE_BODY_4 = ResponseBody.create(MediaType.parse(contentType), content1);
+    private static final Response<Account> ERROR_RESPONSE_4 = Response.error(422, RESPONSE_BODY_4);
+
+    private static HttpException httpException = new HttpException(ERROR_RESPONSE);
+    private static HttpException httpException1 = new HttpException(ERROR_RESPONSE_1);
+
+    private static IOException ioException = new IOException();
+
+    @Test
+    public void shouldReturnNullOnNullAnsEmptyPointedField() throws IOException, URISyntaxException {
+        assertNull(ErrorUtils.getPointedField(null));
+        assertNull(ErrorUtils.getPointedField(""));
+    }
+
+    @Test
+    public void shouldReturnPointedField() {
+        String pointer = "/data/attributes/form/end_field";
+        String pointer1 = "/data/attributes/form_field";
+        String pointer2 = "/data/attributes/";
+        String pointer3 = "/data";
+
+        assertEquals("end_field", ErrorUtils.getPointedField(pointer));
+        assertEquals("form_field", ErrorUtils.getPointedField(pointer1));
+        assertNull(ErrorUtils.getPointedField(pointer2));
+        assertNull(ErrorUtils.getPointedField(pointer3));
+    }
+
+    @Test
+    public void shouldReturnErrorDetailsWithPointedFieldSuccessfully() {
+        String str = String.valueOf(ErrorUtils.getErrorDetails(ERROR_RESPONSE_4.errorBody()));
+
+        assertEquals("Missing data for required field: licence", str);
+        assertNotEquals("Some random string", str);
+    }
+
+    @Test
+    public void shouldReturnErrorDetailsWithNullPointedFieldSuccessfully() {
+        String str = String.valueOf(ErrorUtils.getErrorDetails(ERROR_RESPONSE_2.errorBody()));
+
+        assertEquals("Missing data for required field.", str);
+        assertNotEquals("Some random string", str);
+    }
+
+    @Test
+    public void shouldReturnErrorDetailsWithEmptyPointedFieldSuccessfully() {
+        String str = String.valueOf(ErrorUtils.getErrorDetails(ERROR_RESPONSE_3.errorBody()));
+
+        assertEquals("Missing data for required field.", str);
+        assertNotEquals("Some random string", str);
+    }
+
+    @Test
+    public void shouldReturnErrorMessageSuccessfully() {
+        String str = String.valueOf(ErrorUtils.getMessage(httpException1));
+
+        assertEquals("Missing data for required field: licence", str);
+    }
+
+    @Test
+    public void shouldReturnStoredErrorMessageSuccessfully() {
+        String str = String.valueOf(ErrorUtils.getMessage(httpException));
+
+        assertEquals("Something went wrong! Please check any empty field of a form.", str);
+    }
+
+    @Test
+    public void shouldReturnOtherThrowableErrorMessageSuccessfully() {
+        String str = String.valueOf(ErrorUtils.getMessage(ioException));
+
+        assertEquals("null", str);
```

Comment:
Why is return string "null"

Reply:
The `IOException` thrown in the test has empty message body so `throwable.getMessage()` returns `"null"` but in the application this will most likely not be the case because the system won't throw an empty IOException. I just wanted to highlight the case that's why wrote this test.